client/app/bundles/course/assessment/pages/AssessmentIndex/Actions.tsx

```diff
@@ -0,0 +1,122 @@
+import { Create, Inventory, Lock, QuestionMark } from '@mui/icons-material';
+import { Button, IconButton, Tooltip, Typography } from '@mui/material';
+import { AssessmentListData } from 'types/course/assessment/assessments';
+
+import useTranslation, { Descriptor } from 'lib/hooks/useTranslation';
+
+import translations from './translations';
+
+const ACTION_LABELS: Record<AssessmentListData['status'], Descriptor> = {
+  attempting: translations.resume,
+  locked: translations.unlock,
+  open: translations.attempt,
+  submitted: translations.view,
+  unavailable: translations.attempt,
+};
+
+interface ActionsProps {
+  for: AssessmentListData;
+  student: boolean;
+}
+
+const Actions = (props: ActionsProps): JSX.Element => {
+  const { t } = useTranslation();
+
+  const renderUnavailableMessage = (
```

Comment:
You can move this out from the Actions component.

Reply:
I cannot move `renderUnavailableMessage` out because it uses `t` from `useTranslation`.

Reply:
I decided to make `renderUnavailableMessage` into its own component, just like how it should have been—just realised it when I was doing `AssessmentShow`.

This way, we benefit from React's rendering optimisations magic, no wasted function generations (which I believe was your concern?), and no undeclarative `renderSomething` functions (my pet peeve).

---

P.S. The reason why I do not like `renderSomething` is because it makes the code read weird. One will have to go through a bunch of `render*`s before the actual JSX is returned, then as one reads through, scrolls up to find them weird `render*`s. If it was done for a good reason, e.g., `useMemo`, etc., then it is fine. But then again, I myself have also been guilty of this anti-pattern. Good catch!